tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,157 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"testing"
+	"time"
+)
+
+// TODO: A neater way to provide ocdev path. Currently we assume \
+// ocdev and oc in $PATH already.
+
+func runCmd(cmdS string) string {
+	cmd, err := exec.Command("/bin/sh", "-c", cmdS).Output()
+	if err != nil {
+		log.Fatalf("Error running command %s: %s", cmdS, err)
+		os.Exit(1)
+	}
+	return string(cmd)
+}
+
+func createApp(appName string) {
+	runCmd("ocdev application create " + appName)
+}
+
+func getApp() string {
+	app := runCmd("ocdev application get -q")
+	return app
+}
+
+func getCmp() string {
+	cmp := runCmd("ocdev component get --short")
+	return cmp
+}
+
+func pingSvc(url string) {
+	var ep int
+	pingTimeout := time.After(10 * time.Minute)
+	tick := time.Tick(time.Second)
+
+	for {
+		select {
+		case <-pingTimeout:
+			fmt.Errorf("could not ping the specific service in given time: 10 minutes")
+			os.Exit(1)
+
+		case <-tick:
+			httpTimeout := time.Duration(10 * time.Second)
+			client := http.Client{
+				Timeout: httpTimeout,
+			}
+
+			response, err := client.Get(url)
+			if err != nil {
+				time.Sleep(1 * time.Second)
+				continue
+			}
+
+			if response.Status == "200 OK" {
+				ep = 1
+
+			} else {
+				fmt.Errorf("for service")
+				os.Exit(1)
+			}
+		}
+
+		if ep > 0 {
+			break
+		}
+	}
+}
+
+func TestOCdev(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "ocdev test suite")
+}
+
+var _ = BeforeSuite(func() {
+	runCmd("oc cluster down")
+	runCmd("oc cluster up")
+})
+
+var _ = Describe("Usecase #5", func() {
+
+	TmpDir, err := ioutil.TempDir("", "ocdev")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	It("create application", func() {
+		createApp("usecase5")
+		Expect(getApp()).To(Equal("usecase5"))
+	})
+
+	It("create component", func() {
+		runCmd("git clone https://github.com/openshift/nodejs-ex " + TmpDir + "/nodejs-ex")
+		runCmd("ocdev component create nodejs --dir " + TmpDir + "/nodejs-ex")
+		Expect(getCmp()).To(Equal("nodejs"))
+		time.Sleep(10)
+	})
+
+	It("push changes to component", func() {
+
+		// Create route
+		getIP := runCmd("oc get svc --no-headers | awk '{ print $2 }' | tr -d '\n'")
```

Comment:
you can get the whole url with one command without needing `awk` or `tr`.
`oc` has support for go templates
```
oc get svc nodejs -o go-template="{{.spec.clusterIP}}:{{(index .spec.ports 0).port}}"
```